### ISSUE TITLE
Array props not causing re-render

### ISSUE THREAD

**Issue by bhishp** [meteor/react-packages#222]
The HoC returned by createContainer doesn't seem to do a deep comparison/support array items. Passing an array of objects as props to a meteor container doesn't cause a re-render of the Enhanced Component. i.e.

```
<MeteorEnhanceComponent data={this.state.data} />
// where this.state.data = [ { a: 'foo', b: 'boo'}, { a: 'foobar', b: 'baz' } ]
```

**Problem example**

I have a component that allows a user to select multiple languages from a long list of languages. I pass the currently selected languages down to the `LanguagesSelector` as props, whilst fetching the list of available languages from a Meteor publication.

Container:

```
export default createContainer((props) => {
  Meteor.subscribe('refLanguages');
  let refLanguages = RefLanguages.find().fetch();
  return {
    dataSource: refLanguages,
    ...props, // included in props is props.selectedLanguages
  };
}, LanguagesSelector);
```

From the outer component I control the state of selected languages:

```
setSelectedLanguages(selectedLanguages) {
    this.setState({ selectedLanguages });
}

// in render
<LanguagesSelectorContainer
  selectedData={this.state.selectedLanguages}
  setSelectedData={this.setSelectedLanguages}
/>
```

selectedLanguages is an array of objects as such:
```
this.state = {
  selectedLanguages: [
    {
      key: 0,
      label: 'English',
    },
    {
      key: 1,
      label: 'Urdu',
    },
  ]
};
```

From within LanguagesSelector I call this.props.setSelectedData, e.g.
```
this.props.setSelectedData([{ key: 0, label: 'English' }]
```
I would expect the LanguagesSelector to re-render at this point but it doesn't.

It seems to be something to do with the array/not doing a deep comparison because I can force the re-render by adding a simple Random number as a prop. i.e.

```
// same a s before but generate a random number on every update and pass this as props as well
setSelectedLanguages(selectedLanguages) {
    const rand = Math.random();
    this.setState({ selectedLanguages, rand });
}

// in render
<LanguagesSelectorContainer
  selectedData={this.state.selectedLanguages}
  setSelectedData={this.setSelectedLanguages}
  rand={this.state.rand}
/>
```
Now the LanguagesSelector re-renders on every LanguageUpdate, seemingly because the rand prop is mutating everytime it is called.

**Comment by bhishp** [comment]
Actually, it seems not to be a problem for arrays but for an array of objects. I have modified selectedLanguages to be just an array of those languages and the component re-renders as expected.

`this,state= { selectedLanguages: ['English', 'Urdu'] }`

**Comment by klaussner** [comment]
I wasn't able to reproduce the issue with the code you posted but it sounds like it could have to do with the fact that [`createContainer` creates a pure component by default](https://github.com/meteor/react-packages/blob/cead57f6a3164ec22e42aff404111c6f057b0e9b/packages/react-meteor-data/ReactMeteorData.jsx#L3). Does the problem still occur if you create the container like this?

```js
export default createContainer({
  getMeteorData: (props) => {
    Meteor.subscribe('refLanguages');
    ...
  },
  pure: false
}, LanguagesSelector);
```

**Comment by bhishp** [comment]
I have just re-visited this (I got around it before by just using a flat array of `languages = ['Urdu', 'English']`) And I am also unable to re-create for myself :-|

I have changed a couple of things since I first posted so can't tell exactly what it was that may have affected. But, maybe I had something out of place before, because even now with an array of objects again it is re-rendering as expected. To confirm, I have *not* have to set `pure: false` and it still does work.
